src/objecttypes/api/views.py

```diff
@@ -1,13 +1,42 @@
-from rest_framework import viewsets
+from drf_yasg.utils import no_body, swagger_auto_schema
+from rest_framework import mixins, viewsets
+from rest_framework.decorators import action
+from rest_framework.response import Response
+from vng_api_common.viewsets import NestedViewSetMixin
 
-from objecttypes.core.models import ObjectType
+from objecttypes.core.constants import ObjectVersionStatus
+from objecttypes.core.models import ObjectType, ObjectVersion
 
 from .filters import ObjectTypeFilterSet
-from .serializers import ObjectTypeSerializer
+from .serializers import ObjectTypeSerializer, ObjectVersionSerializer
 
 
-class ObjectTypeViewSet(viewsets.ReadOnlyModelViewSet):
+class ObjectTypeViewSet(
+    mixins.CreateModelMixin, mixins.UpdateModelMixin, viewsets.ReadOnlyModelViewSet
+):
     queryset = ObjectType.objects.prefetch_related("versions").order_by("-pk")
     serializer_class = ObjectTypeSerializer
     lookup_field = "uuid"
     filterset_class = ObjectTypeFilterSet
+
+
+class ObjectVersionViewSet(
+    NestedViewSetMixin,
+    mixins.CreateModelMixin,
+    mixins.UpdateModelMixin,
+    viewsets.ReadOnlyModelViewSet,
+):
+    queryset = ObjectVersion.objects.order_by("object_type", "-version")
+    serializer_class = ObjectVersionSerializer
+    lookup_field = "version"
+
+    @swagger_auto_schema(request_body=no_body)
+    @action(detail=True, methods=["post"])
+    def publish(self, request, *args, **kwargs):
```

Comment:
I understand that this is an action BUT we have 3 statusses: draft, published and deprecated. There is no `status` field that can be modified on a version object. So, how do we mark a version as deprecated?

I think we should just add the status field to a writable field so you can set published like that, and also set deprecated. We should only block changing these 2 back to draft.

Reply:
Actually I quite like `publish` endpoint, it makes API consistent with admin page, and also with our Catalogi API. 

Do we really need `deprecated` status? We don't use it anywhere now. Maybe we can just get rid of it.
Or we can add DELETE action to the Version resource and switch status to `depricated` in this case?

Reply:
> Actually I quite like publish endpoint, it makes API consistent with admin page

That's not really a good reason. The admin can just as well set the status field (in fact, it does that ofcourse, but via a publish function).

> and also with our Catalogi API.

Hmm, that's a valid argument: Consistency.
Since we disagree, I'll ask a third party :)

> Do we really need deprecated status? We don't use it anywhere now. 

It was added because of DCAT/Metadata catalogus

> Or we can add DELETE action to the Version resource and switch status to depricated in this case?

No, this is unexpected behaviour. Plus, if we ever introduce a new status it will be weird.

Reply:
I like the idea of a `DELETE` call:

* on draft resources -> actual DB delete
* on published resources -> mark deprecated. Mistakes can then be correct be publishing it again, if need be?

Reply:
I don't like that "draft" is set automatically, "publish" is set by an action and "deprecated" is set with a HTTP method.

1. I'll make the decision to set the status with PUT/PATCH only. Setting non-draft to draft is not allowed.
2. DELETE should be allowed on versions that are in draft.
3. DELETE should also be allowed on objecttypes that have no versions

Reply:
Ok, I'll change the implementation

Reply:
the implementation is changed:
* DELETE is supported now
* "publish" action is removed